modules/social_features/social_embed/src/Plugin/Filter/SocialEmbedLazyFilter.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Drupal\social_embed\Plugin\Filter;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\filter\FilterProcessResult;
+use Drupal\filter\Plugin\FilterBase;
+
+/**
+ * Provides a filter to lazy-load images.
+ *
+ * @Filter(
+ *   id = "social_embed_lazy_filter",
+ *   title = @Translation("Lazy-load images and iframes"),
+ *   description = @Translation("Only selected tags will be lazy-loaded in activated text-formats."),
+ *   type = Drupal\filter\Plugin\FilterInterface::TYPE_TRANSFORM_REVERSIBLE,
+ *   settings = {
+ *     "image" = TRUE,
+ *     "iframe" = TRUE,
+ *   },
+ *   weight = 20
+ * )
+ */
+class SocialEmbedLazyFilter extends FilterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $form['info'] = [
+      '#markup' => $this->t('Lazy-load filter can be enabled for images and iframes.'),
+    ];
+    $form['image'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable for images (%img tags)', ['%img' => '<img>']),
+      "#description" => $this->t('This option only applies to inline-images. If <em>Embed media</em> filter is enabled, the images embedded from media library would use the the selected view mode settings.'),
+      '#default_value' => $this->settings['image'],
+      '#return_value' => TRUE,
+    ];
```

Comment:
I don't think we need image; I think we use Drupal Core's native one for that (see https://github.com/goalgorilla/open_social/pull/3555) so potentially we don't even need this setting and just have it enabled for iFrames whenever there is an iFrame added